sqlalchemy_ibmi/base.py

```diff
@@ -753,15 +753,12 @@ def _parse_dbapi_version(self, vers):
         return vers
 
     def _get_server_version_info(self, connection, allow_chars=True):
-        # NOTE: this function is not reliable, particularly when
-        # freetds is in use.   Implement database-specific server version
-        # queries.
-        dbapi_con = connection.connection
-        version = []
-        r = re.compile(r"[.\-]")
-        for n in r.split(dbapi_con.getinfo(self.dbapi.SQL_DBMS_VER)):
-            version.append(int(n))
-        return tuple(version)
+        if getattr(connection, "connection"):
+            dbapi_con = connection.connection
+            version = dbapi_con.getinfo(self.dbapi.SQL_DBMS_VER).split('.')
```

Comment:
```suggestion
            version = [int(_) for _ in dbapi_con.getinfo(self.dbapi.SQL_DBMS_VER).split('.')]
```

Need to convert to integers yet.